File: android/multando-sdk/src/main/kotlin/com/multando/sdk/models/Evidence.kt
package com.multando.sdk.models

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class EvidenceCreate(
    @SerialName("report_id") val reportId: String,
    @SerialName("evidence_type") val evidenceType: EvidenceType,
    @SerialName("file_url") val fileUrl: String,
    val description: String? = null
)

@Serializable
data class EvidenceResponse(
    val id: String,
    @SerialName("report_id") val reportId: String,
    @SerialName("evidence_type") val evidenceType: EvidenceType,
    @SerialName("file_url") val fileUrl: String,
    val description: String? = null,
    @SerialName("created_at") val createdAt: String
)

File: ios/Sources/MultandoSDK/Models/Evidence.swift
import Foundation

/// Payload for attaching evidence to a report.
public struct EvidenceCreate: Codable, Sendable {
    public let reportId: String
    public let evidenceType: EvidenceType
    public let fileUrl: String
    public let description: String?

    public init(reportId: String, evidenceType: EvidenceType, fileUrl: String, description: String? = nil) {
        self.reportId = reportId
        self.evidenceType = evidenceType
        self.fileUrl = fileUrl
        self.description = description
    }

    enum CodingKeys: String, CodingKey {
        case reportId = "report_id"
        case evidenceType = "evidence_type"
        case fileUrl = "file_url"
        case description
    }
}

/// Evidence record returned by the API.
public struct EvidenceResponse: Codable, Sendable {
    public let id: String
    public let reportId: String
    public let evidenceType: EvidenceType
    public let fileUrl: String
    public let description: String?
    public let createdAt: String

    enum CodingKeys: String, CodingKey {
        case id
        case reportId = "report_id"
        case evidenceType = "evidence_type"
        case fileUrl = "file_url"
        case description
        case createdAt = "created_at"
    }
}
